ceph/rados/pool_workflows.py

```diff
@@ -115,13 +117,70 @@ def fill_omap_entries(self, pool_name, **kwargs):
         # removing the py file copied
         client_node.exec_command(sudo=True, cmd="rm -rf generate_omap_entries.py")
 
-        # restarting primary osd as metadata does not get updated automatically
-        # when number of object containing omap entries is below ~40
-        # this a workaround and will be addressed when RCA is complete
-        primary_osd = self.rados_obj.get_pg_acting_set(pool_name=pool_name)[0]
-        self.rados_obj.change_osd_state(action="restart", target=primary_osd)
+        # Storing the pg dump log before setting the scrub parameters
+        before_scrub_log = self.scrub_obj.get_pg_dump("pgid", "last_scrub_stamp")
+        before_deep_scrub_log = self.scrub_obj.get_pg_dump(
+            "pgid", "last_deep_scrub_stamp"
+        )
+
+        # Triggering scrub & deep scrub on the pool
+        self.rados_obj.run_scrub(pool=pool_name)
+        time.sleep(2)
+        self.rados_obj.run_deep_scrub(pool=pool_name)
+        log.debug(
+            "Sleeping for 20 seconds for scrubbing to complete and proceeding to varify the same"
+        )
+        time.sleep(20)
```

Comment:
If we are already going to smart wait for 50 mins for scrub and deep-scrub to complete, is it necessary to have this blind sleep?

Reply:
Can be removed. Updating code...